### PR TITLE
pixman: 0.38.4 -> 0.42.2

### PR DESCRIPTION
**Pull request by mweinelt**
http://www.openwall.com/lists/oss-security/2022/11/05/1

Fixes: CVE-2022-44638
Backport: #199676

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vcunat** [comment]
openwall says
> The request has failed: No such message

**Comment by mweinelt** [comment]
> openwall says
> > The request has failed: No such message 

https://www.openwall.com/lists/oss-security/2022/11/05/1

File: pkgs/development/libraries/pixman/default.nix [hunk 1]
```diff
@@ -2,11 +2,14 @@
...
   pname = "pixman";
-  version = "0.38.4";
+  version = "0.42.2";
 
   src = fetchurl {
-    url = "mirror://xorg/individual/lib/${pname}-${version}.tar.bz2";
-    sha256 = "0l0m48lnmdlmnaxn2021qi5cj366d9fzfjxkqgcj9bs14pxbgaw4";
+    urls = [
+      "mirror://xorg/individual/lib/${pname}-${version}.tar.bz2"
+      "https://cairographics.org/releases/${pname}-${version}.tar.gz"
```
Comment (vcunat) [inline pkgs/development/libraries/pixman/default.nix:10]: Wait, don't tell me that `.tar.gz` has the same hash as `.tar.bz2`!

Comment (vcunat) [reply pkgs/development/libraries/pixman/default.nix:10]: `cairographics.org` only has `.gz`

Comment (vcunat) [reply pkgs/development/libraries/pixman/default.nix:10]: xorg has only `.gz` and `.xz` for the newer pixman releases.

Comment (vcunat) [reply pkgs/development/libraries/pixman/default.nix:10]: amended the URL

**Comment by vcunat** [comment]
My understanding is that this is a backwards compatible update, despite the larger change in version number.

**Comment by vcunat** [comment]
Unfortunately it broke on `aarch64-darwin`, apparently with some assembler incompatibility: https://hydra.nixos.org/build/197830077

**Comment by vcunat** [comment]
I expect the aarch64 assembly needs GNU instead of LLVM: https://gitlab.freedesktop.org/pixman/pixman/-/issues/46#note_1436787

**Comment by vcunat** [comment]
Let's try https://github.com/NixOS/nixpkgs/pull/200156

**Pull request merged**